config/config_test.go

```diff
@@ -11,24 +11,28 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package config
+package config_test
 
 import (
 	"encoding/json"
 	"io/ioutil"
 	"net/url"
 	"path/filepath"
 	"regexp"
+	"sort"
 	"strings"
 	"testing"
 	"time"
 
+	"github.com/google/go-cmp/cmp"
```

Comment:
Do we really need this dependency? It's not what we already use for comparing structs, and is only for the sake of a unittest for something we don't support.

Reply:
It's used to compare configs after roundtrip-ing them through yaml. This causes the secrets to be masked and the existing comparison mechanism doesn't seem to be able to take that into account.

Reply:
Roundtrip-ing the yaml is to test marshaling. There were essentially no tests for marshaling (just one to make sure secrets were masked). Since there's a lot of effort behind testing the unmarshaling a large "good" config, this lets the marshal test piggyback on that for free.

Reply:
This dependency is still here. If you want to test this, do so without adding a 2nd way to do struct comparisons to the codebase.